src/components/add-journey/JourneySearchResults.tsx

```diff
@@ -13,11 +14,29 @@ const generateJourneyKey = (journey: Journey) => {
   return `${journey.from.departureTimestamp}${journey.from.departure}${journey.to.arrivalTimestamp}${journey.to.arrival}`;
 };
 
+const sortJourneys = (a: Journey, b: Journey) => {
+  return a.from.departureTimestamp - b.from.departureTimestamp;
+};
+
+const unionJourneys = (journeys: Journey[], newJourneys: Journey[]) => {
+  const concatJourneys = journeys.concat(newJourneys);
+  const filteredJourneys = concatJourneys.filter(
+    (item, idx) =>
+      concatJourneys.map((journey) => journey.from.departureTimestamp).indexOf(item.from.departureTimestamp) === idx
+  );
+  return filteredJourneys.sort(sortJourneys);
+};
+
```

Comment:
fiddled around with different ways to do this set union and ended up getting into issues where the Journey `key` was duplicated due to duplicates not being detected. This seemed like the most effective way to ensure that I didn't get copies of journeys with the same departure time.